### ISSUE TITLE
grpc "Failed to connect to all addresses"

### ISSUE THREAD

**Issue by gittripley** [grpc/grpc#27396]
### What version of gRPC and what language are you using?
I have JavaScript gPRC server  @grpc/grpc-js@1.3.7 and Python client 1.40.0

### What operating system (Linux, Windows,...) and version?
Docker Ubuntu 20, both server and client are in the same docker.

I deployed to several machines. One of them consistent fail, while other works. 

### What runtime / compiler are you using (e.g. python version or version of gcc)
python version

### What did you do?
When make remote call using python client, I got the following errors

`"description":"Failed to pick subchannel","file":"src/core/ext/filters/client_channel/client_channel.cc","file_line":3186,"referenced_errors":[{"created":"@1631934123.441737107","description":"failed to connect to all addresses","file":"src/core/lib/transport/error_utils.cc","file_line":146,"grpc_status":14` 

I searched around, other people's error is different. not from `error_utils.cc` but from `pick_first.cc`

It is always reproducible.

**Comment by yashykt** [comment]
From the description, I'm suspecting a setup issue rather than a gRPC issue, but if you have a reproduction that proves that it has nothing to do with your specific setup, we'll take a look.

**Comment by gittripley** [comment]
@yashykt 
Thanks for the response. Finally, I figured out. I have a subprocess to start the JavaScript GRPC server and then make a client call right after it. Most of machine works, but one machine throw the error above. It turns out that when client connect to server before the server is ready. Later, I changed the code to wait a few second before sending the request to server, or keep checking whether the server is ready. I never had the problem on that machine.
